vector/src/Data/Vector/Primitive.hs

```diff
@@ -1313,9 +1315,9 @@ foldMap' = G.foldMap'
 -- ==== __Examples__
 --
 -- >>> import qualified Data.Vector.Primitive as VP
--- >>> VP.all even $ VP.fromList [2, 4, 12 :: Int]
```

Comment:
I'd rather not remove type annotations from examples for `Primitive` and `Unboxed` vectors. This is because the `$setup` is not visible in haddock and I would like any novice user to be able to copy and paste examples directly from haddock and be able and run them instead of getting a cryptic message from ghci.

Reply:
As I suggested in https://github.com/haskell/vector/pull/398#discussion_r685596220, we could just mention that all the examples use `Int`.

Reply:
We could surely mention it, but who would imagine "all examples use `Int`" means "If you would like to try these examples yourself in `ghci`, you would need to add `:: Int` somewhere in the expressions," instead of "Due to our laziness, all examples we prepared will be inferred for type `Int` by GHC"?

...The previous paragraph was a nitpicking, but honestly I'd like to ask: is there any reason `::Int` should be dropped other than aesthetic simplicity? I guess one of the best things about doctests is that beginners can themselves test the expression on GHCi. It would be best to have the examples self-contained in each function; I wouldn't like beginners to waste their minutes or hours wondering why the examples don't work until they realize that they need to climb up to the beginning of the documentation.

Reply:
> until they realize that they need to climb up to the beginning of the documentation.

@gksato It is even worse than that. `$setup` is not even included in the generated haddock. So users will not even be able to find it in the documentation why these doctests don't work for them.

@konsumlamm That is exactly why I am opposed having some magic in doctests just to make them a bit prettier